### PR TITLE
Avoid NullPointerException describing null value

### LINKED ISSUE #171: NullPointerException at JSONCompareResult.describe
If expected or actual value in DefaultComparator.compareValues(...) is null then a NullPointerException will be thrown by formatting of message.

org.skyscreamer:jsonassert:1.5.1

java.lang.NullPointerException: Cannot invoke "Object.toString()" because "value" is null

	at org.skyscreamer.jsonassert.JSONCompareResult.describe(JSONCompareResult.java:248)
	at org.skyscreamer.jsonassert.JSONCompareResult.formatFailureMessage(JSONCompareResult.java:200)
	at org.skyscreamer.jsonassert.JSONCompareResult.fail(JSONCompareResult.java:180)
	at org.skyscreamer.jsonassert.comparator.DefaultComparator.compareValues(DefaultComparator.java:57)
	at org.skyscreamer.jsonassert.comparator.CustomComparator.compareValues(CustomComparator.java:48)

Proposal: Replace lines (in DefaultComparator.compareValues(...))

```
        if ((expectedValue == null && actualValue != null) || (expectedValue != null && actualValue == null)) {
            result.fail(prefix, expectedValue, actualValue);
        }
```
to

```
        if (expectedValue == null && actualValue != null) {
            result.unexpected(prefix, actualValue);
        }
        if (expectedValue != null && actualValue == null) {
            result.missing(prefix, expectedValue);
        }
```

### PR DESCRIPTION
**Pull request by MikeEdgar**
Fixes #171

### CODE REVIEW

File: src/main/java/org/skyscreamer/jsonassert/comparator/DefaultComparator.java [hunk 1]
```diff
@@ -53,10 +53,9 @@ public void compareValues(String prefix, Object expectedValue, Object actualValu
         if (expectedValue == actualValue) {
             return;
         }
-        if ((expectedValue == null && actualValue != null) || (expectedValue != null && actualValue == null)) {
+        if (expectedValue == null || actualValue == null) {
             result.fail(prefix, expectedValue, actualValue);
-        }
-        if (areNumbers(expectedValue, actualValue)) {
+        } else if (areNumbers(expectedValue, actualValue)) {
```
Comment (MikeEdgar) [inline src/main/java/org/skyscreamer/jsonassert/comparator/DefaultComparator.java:58]: Removed extra "always true" non-null checks and do not attempt to fail for other reasons when already failing due to one value being null.

**Comment by MikeEdgar** [comment]
@carterpage , please take a look.

**Comment by carterpage** [comment]
Thanks for the ping. Looks good. I'll try to get this merged and released this weekend.

**Comment by davenfroberg** [comment]
@carterpage status on this PR? fixes an issue I'm having (#171)

**Comment by Azquelt** [comment]
I hit this too and was preparing to submit a similar PR.

**Comment by carterpage** [comment]
I have been getting new releases going again, and will try to get this included in the new v2 in the next couple of weeks.

**Pull request merged**